modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -332,7 +332,6 @@ private void restart() throws MuleException {
     synchronized (started) {
       if (started.get()) {
         stopSource();
-        disposeSource();
```

Comment:
removing this is a no go. dispose() phase MUST Be called on the source when restarting